app/views/sentences/several_random_sentences.ctp

```diff
@@ -90,21 +89,43 @@ $this->set('title_for_layout', $pages->formatTitle(__('Random sentences', true))
     </div>
 </div>    
     
-    
-<div id="main_content">
-    <div class="module">
-    <h2><?php __('Random sentences'); ?></h2>
-        <?php
-        foreach ($allSentences as $index=>$sentence) {
-            $sentences->displaySentencesGroup(
-                $sentence['Sentence'],
-                $sentence['Transcription'],
-                $sentence['Translation'],
-                $sentence['User']
-            );
+<?php
+if(!isset($searchProblem)) {
+?>  
+    <div id="main_content">
+        <div class="module">
+        <?php  
+            $this->Html->tag('h2', 'Random sentences', null);
```

Comment:
Up to here, the code can be taken out of the `if()` block to avoid duplication with the corresponding `else`.